src/Microsoft.OData.Core/MediaTypeUtils.cs

```diff
@@ -934,11 +934,34 @@ public MatchInfoCacheKey(ODataMediaTypeResolver resolver, ODataPayloadKind paylo
             /// <returns>true if obj is equal to this instance; otherwise, false.</returns>
             public override bool Equals(object obj)
             {
-                MatchInfoCacheKey cacheKey = obj as MatchInfoCacheKey;
-                return cacheKey != null &&
-                    this.MediaTypeResolver == cacheKey.MediaTypeResolver &&
-                    this.PayloadKind == cacheKey.PayloadKind &&
-                    this.ContentTypeName == cacheKey.ContentTypeName;
+                return this.Equals(obj as MatchInfoCacheKey);
+            }
+
+            /// <summary>
+            /// Returns a value indicating whether this instance is equal to a specified <see cref="MatchInfoCacheKey"/>.
+            /// </summary>
+            /// <param name="other">An object to compare with this instance.</param>
+            /// <returns>true if obj is equal to this instance; otherwise, false.</returns>
+            public bool Equals(MatchInfoCacheKey other)
+            {
+                if (ReferenceEquals(other, null))
+                {
+                    return false;
+                }
+
+                if (ReferenceEquals(this, other))
+                {
+                    return true;
+                }
+
+                if (this.GetType() != other.GetType())
+                {
+                    return false;
+                }

```

Comment:
What is this checking for? When would the types be different?

Reply:
You are right. It's unnecessary. 

---
In reply to: [285791784](https://github.com/OData/odata.net/pull/1448#discussion_r285791784) [](ancestors = 285791784)